App add auth routes

TurpaultM opened the pull request:
auth routes for app
register
login
logout
forgot pwd
reset pwd

TurpaultM committed:
ADD: auth routes

TurpaultM committed:
Merge branch 'dev_v2' of github.com:Jessika63/PharmaXcess into APP_add_auth_routes

TurpaultM committed:
Merge branch 'dev_v2' of github.com:Jessika63/PharmaXcess into APP_add_auth_routes

TurpaultM committed:
UPDATE: fix auth routes

ErwannL commented:
i get those errors:
- when i do forgot password:
on postman
```html
<!doctype html>
<html lang=en>

<head>
    <title>pymysql.err.OperationalError: (1054, &#34;Unknown column &#39;reset_token&#39; in &#39;field list&#39;&#34;)
        // Werkzeug Debugger</title>
    <link rel="stylesheet" href="?__debugger__=yes&amp;cmd=resource&amp;f=style.css">
    <link rel="shortcut icon" href="?__debugger__=yes&amp;cmd=resource&amp;f=console.png">
    <script src="?__debugger__=yes&amp;cmd=resource&amp;f=debugger.js"></script>
    <script>
        var CONSOLE_MODE = false,
          EVALEX = true,
          EVALEX_TRUSTED = false,
          SECRET = "aRo3kbqjFvWGvZZ1gSRk";
    </script>
</head>

<body style="background-color: #fff">
    <div class="debugger">
        <h1>OperationalError</h1>
        <div class="detail">
            <p class="errormsg">pymysql.err.OperationalError: (1054, &#34;Unknown column &#39;reset_token&#39; in
                &#39;field list&#39;&#34;)
            </p>
        </div>
        <h2 class="traceback">Traceback <em>(most recent call last)</em></h2>
        <div class="traceback">
            <h3></h3>
            <ul>
                <li>
                    <div class="frame" id="frame-140118917210848">
                        <h4>File <cite class="filename">"/usr/local/lib/python3.9/site-packages/flask/app.py"</cite>,
                            line <em class="line">1536</em>,
                            in <code class="function">__call__</code></h4>
                        <div class="source library">
                            <pre class="line before"><span class="ws">    </span>) -&gt; cabc.Iterable[bytes]:</pre>
                            <pre
                                class="line before"><span class="ws">        </span>&#34;&#34;&#34;The WSGI server calls the Flask application object as the</pre>
                            <pre
                                class="line before"><span class="ws">        </span>WSGI application. This calls :meth:`wsgi_app`, which can be</pre>
                            <pre class="line before"><span class="ws">        </span>wrapped to apply middleware.</pre>
                            <pre class="line before"><span class="ws">        </span>&#34;&#34;&#34;</pre>
                            <pre
                                class="line current"><span class="ws">        </span>return self.wsgi_app(environ, start_response)</pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916568112">
                        <h4>File <cite class="filename">"/usr/local/lib/python3.9/site-packages/flask/app.py"</cite>,
                            line <em class="line">1514</em>,
                            in <code class="function">wsgi_app</code></h4>
                        <div class="source library">
                            <pre class="line before"><span class="ws">            </span>try:</pre>
                            <pre class="line before"><span class="ws">                </span>ctx.push()</pre>
                            <pre
                                class="line before"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
                            <pre class="line before"><span class="ws">            </span>except Exception as e:</pre>
                            <pre class="line before"><span class="ws">                </span>error = e</pre>
                            <pre
                                class="line current"><span class="ws">                </span>response = self.handle_exception(e)</pre>
                            <pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
                            <pre
                                class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre>
                            <pre class="line after"><span class="ws">                </span>raise</pre>
                            <pre
                                class="line after"><span class="ws">            </span>return response(environ, start_response)</pre>
                            <pre class="line after"><span class="ws">        </span>finally:</pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916568224">
                        <h4>File <cite class="filename">"/usr/local/lib/python3.9/site-packages/flask/app.py"</cite>,
                            line <em class="line">1511</em>,
                            in <code class="function">wsgi_app</code></h4>
                        <div class="source library">
                            <pre
                                class="line before"><span class="ws">        </span>ctx = self.request_context(environ)</pre>
                            <pre
                                class="line before"><span class="ws">        </span>error: BaseException | None = None</pre>
                            <pre class="line before"><span class="ws">        </span>try:</pre>
                            <pre class="line before"><span class="ws">            </span>try:</pre>
                            <pre class="line before"><span class="ws">                </span>ctx.push()</pre>
                            <pre
                                class="line current"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
                            <pre class="line after"><span class="ws">            </span>except Exception as e:</pre>
                            <pre class="line after"><span class="ws">                </span>error = e</pre>
                            <pre
                                class="line after"><span class="ws">                </span>response = self.handle_exception(e)</pre>
                            <pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
                            <pre
                                class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916568336">
                        <h4>File <cite class="filename">"/usr/local/lib/python3.9/site-packages/flask/app.py"</cite>,
                            line <em class="line">919</em>,
                            in <code class="function">full_dispatch_request</code></h4>
                        <div class="source library">
                            <pre
                                class="line before"><span class="ws">            </span>request_started.send(self, _async_wrapper=self.ensure_sync)</pre>
                            <pre
                                class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
                            <pre class="line before"><span class="ws">            </span>if rv is None:</pre>
                            <pre
                                class="line before"><span class="ws">                </span>rv = self.dispatch_request()</pre>
                            <pre class="line before"><span class="ws">        </span>except Exception as e:</pre>
                            <pre
                                class="line current"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
                            <pre
                                class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                            <pre class="line after"><span class="ws">    </span>def finalize_request(</pre>
                            <pre class="line after"><span class="ws">        </span>self,</pre>
                            <pre
                                class="line after"><span class="ws">        </span>rv: ft.ResponseReturnValue | HTTPException,</pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916568448">
                        <h4>File <cite class="filename">"/usr/local/lib/python3.9/site-packages/flask/app.py"</cite>,
                            line <em class="line">917</em>,
                            in <code class="function">full_dispatch_request</code></h4>
                        <div class="source library">
                            <pre class="line before"><span class="ws"></span> </pre>
                            <pre class="line before"><span class="ws">        </span>try:</pre>
                            <pre
                                class="line before"><span class="ws">            </span>request_started.send(self, _async_wrapper=self.ensure_sync)</pre>
                            <pre
                                class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
                            <pre class="line before"><span class="ws">            </span>if rv is None:</pre>
                            <pre
                                class="line current"><span class="ws">                </span>rv = self.dispatch_request()</pre>
                            <pre class="line after"><span class="ws">        </span>except Exception as e:</pre>
                            <pre
                                class="line after"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
                            <pre
                                class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                            <pre class="line after"><span class="ws">    </span>def finalize_request(</pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916568560">
                        <h4>File <cite class="filename">"/usr/local/lib/python3.9/site-packages/flask/app.py"</cite>,
                            line <em class="line">902</em>,
                            in <code class="function">dispatch_request</code></h4>
                        <div class="source library">
                            <pre
                                class="line before"><span class="ws">            </span>and req.method == &#34;OPTIONS&#34;</pre>
                            <pre class="line before"><span class="ws">        </span>):</pre>
                            <pre
                                class="line before"><span class="ws">            </span>return self.make_default_options_response()</pre>
                            <pre
                                class="line before"><span class="ws">        </span># otherwise dispatch to the handler for that endpoint</pre>
                            <pre
                                class="line before"><span class="ws">        </span>view_args: dict[str, t.Any] = req.view_args  # type: ignore[assignment]</pre>
                            <pre
                                class="line current"><span class="ws">        </span>return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                            <pre
                                class="line after"><span class="ws">    </span>def full_dispatch_request(self) -&gt; Response:</pre>
                            <pre
                                class="line after"><span class="ws">        </span>&#34;&#34;&#34;Dispatches the request and on top of that performs request</pre>
                            <pre
                                class="line after"><span class="ws">        </span>pre and postprocessing as well as HTTP exception catching and</pre>
                            <pre class="line after"><span class="ws">        </span>error handling.</pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916568672">
                        <h4>File <cite class="filename">"/app/routes/app/forgot_password.py"</cite>,
                            line <em class="line">18</em>,
                            in <code class="function">forgot_password</code></h4>
                        <div class="source ">
                            <pre class="line before"><span class="ws">    </span>reset_token = str(uuid.uuid4())</pre>
                            <pre
                                class="line before"><span class="ws">    </span>expiration = datetime.datetime.now() + datetime.timedelta(hours=1)</pre>
                            <pre class="line before"><span class="ws"></span> </pre>
                            <pre class="line before"><span class="ws">    </span>conn = get_app_connection()</pre>
                            <pre class="line before"><span class="ws">    </span>with conn.cursor() as cursor:</pre>
                            <pre class="line current"><span class="ws">        </span>cursor.execute(</pre>
                            <pre
                                class="line after"><span class="ws">            </span>&#34;UPDATE utilisateurs SET reset_token=%s, reset_token_expiration=%s WHERE email=%s&#34;,</pre>
                            <pre
                                class="line after"><span class="ws">            </span>(reset_token, expiration, email)</pre>
                            <pre class="line after"><span class="ws">        </span>)</pre>
                            <pre class="line after"><span class="ws">    </span>conn.commit()</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916569232">
                        <h4>File
                            <cite class="filename">"/usr/local/lib/python3.9/site-packages/pymysql/cursors.py"</cite>,
                            line <em class="line">153</em>,
                            in <code class="function">execute</code></h4>
                        <div class="source library">
                            <pre class="line before"><span class="ws">        </span>while self.nextset():</pre>
                            <pre class="line before"><span class="ws">            </span>pass</pre>
                            <pre class="line before"><span class="ws"></span> </pre>
                            <pre
                                class="line before"><span class="ws">        </span>query = self.mogrify(query, args)</pre>
                            <pre class="line before"><span class="ws"></span> </pre>
                            <pre class="line current"><span class="ws">        </span>result = self._query(query)</pre>
                            <pre class="line after"><span class="ws">        </span>self._executed = query</pre>
                            <pre class="line after"><span class="ws">        </span>return result</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                            <pre
                                class="line after"><span class="ws">    </span>def executemany(self, query, args):</pre>
                            <pre
                                class="line after"><span class="ws">        </span>&#34;&#34;&#34;Run several data against one query.</pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916590384">
                        <h4>File
                            <cite class="filename">"/usr/local/lib/python3.9/site-packages/pymysql/cursors.py"</cite>,
                            line <em class="line">322</em>,
                            in <code class="function">_query</code></h4>
                        <div class="source library">
                            <pre class="line before"><span class="ws">        </span>self.rownumber = r</pre>
                            <pre class="line before"><span class="ws"></span> </pre>
                            <pre class="line before"><span class="ws">    </span>def _query(self, q):</pre>
                            <pre class="line before"><span class="ws">        </span>conn = self._get_db()</pre>
                            <pre class="line before"><span class="ws">        </span>self._clear_result()</pre>
                            <pre class="line current"><span class="ws">        </span>conn.query(q)</pre>
                            <pre class="line after"><span class="ws">        </span>self._do_get_result()</pre>
                            <pre class="line after"><span class="ws">        </span>return self.rowcount</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                            <pre class="line after"><span class="ws">    </span>def _clear_result(self):</pre>
                            <pre class="line after"><span class="ws">        </span>self.rownumber = 0</pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916590496">
                        <h4>File
                            <cite class="filename">"/usr/local/lib/python3.9/site-packages/pymysql/connections.py"</cite>,
                            line <em class="line">563</em>,
                            in <code class="function">query</code></h4>
                        <div class="source library">
                            <pre class="line before"><span class="ws">        </span># if DEBUG:</pre>
                            <pre
                                class="line before"><span class="ws">        </span>#     print(&#34;DEBUG: sending query:&#34;, sql)</pre>
                            <pre class="line before"><span class="ws">        </span>if isinstance(sql, str):</pre>
                            <pre
                                class="line before"><span class="ws">            </span>sql = sql.encode(self.encoding, &#34;surrogateescape&#34;)</pre>
                            <pre
                                class="line before"><span class="ws">        </span>self._execute_command(COMMAND.COM_QUERY, sql)</pre>
                            <pre
                                class="line current"><span class="ws">        </span>self._affected_rows = self._read_query_result(unbuffered=unbuffered)</pre>
                            <pre class="line after"><span class="ws">        </span>return self._affected_rows</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                            <pre
                                class="line after"><span class="ws">    </span>def next_result(self, unbuffered=False):</pre>
                            <pre
                                class="line after"><span class="ws">        </span>self._affected_rows = self._read_query_result(unbuffered=unbuffered)</pre>
                            <pre class="line after"><span class="ws">        </span>return self._affected_rows</pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916739584">
                        <h4>File
                            <cite class="filename">"/usr/local/lib/python3.9/site-packages/pymysql/connections.py"</cite>,
                            line <em class="line">825</em>,
                            in <code class="function">_read_query_result</code></h4>
                        <div class="source library">
                            <pre
                                class="line before"><span class="ws">                </span>result.unbuffered_active = False</pre>
                            <pre
                                class="line before"><span class="ws">                </span>result.connection = None</pre>
                            <pre class="line before"><span class="ws">                </span>raise</pre>
                            <pre class="line before"><span class="ws">        </span>else:</pre>
                            <pre
                                class="line before"><span class="ws">            </span>result = MySQLResult(self)</pre>
                            <pre class="line current"><span class="ws">            </span>result.read()</pre>
                            <pre class="line after"><span class="ws">        </span>self._result = result</pre>
                            <pre
                                class="line after"><span class="ws">        </span>if result.server_status is not None:</pre>
                            <pre
                                class="line after"><span class="ws">            </span>self.server_status = result.server_status</pre>
                            <pre class="line after"><span class="ws">        </span>return result.affected_rows</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916739696">
                        <h4>File
                            <cite class="filename">"/usr/local/lib/python3.9/site-packages/pymysql/connections.py"</cite>,
                            line <em class="line">1199</em>,
                            in <code class="function">read</code></h4>
                        <div class="source library">
                            <pre class="line before"><span class="ws">        </span>if self.unbuffered_active:</pre>
                            <pre
                                class="line before"><span class="ws">            </span>self._finish_unbuffered_query()</pre>
                            <pre class="line before"><span class="ws"></span> </pre>
                            <pre class="line before"><span class="ws">    </span>def read(self):</pre>
                            <pre class="line before"><span class="ws">        </span>try:</pre>
                            <pre
                                class="line current"><span class="ws">            </span>first_packet = self.connection._read_packet()</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                            <pre
                                class="line after"><span class="ws">            </span>if first_packet.is_ok_packet():</pre>
                            <pre
                                class="line after"><span class="ws">                </span>self._read_ok_packet(first_packet)</pre>
                            <pre
                                class="line after"><span class="ws">            </span>elif first_packet.is_load_local_packet():</pre>
                            <pre
                                class="line after"><span class="ws">                </span>self._read_load_local_packet(first_packet)</pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916739808">
                        <h4>File
                            <cite class="filename">"/usr/local/lib/python3.9/site-packages/pymysql/connections.py"</cite>,
                            line <em class="line">775</em>,
                            in <code class="function">_read_packet</code></h4>
                        <div class="source library">
                            <pre class="line before"><span class="ws"></span> </pre>
                            <pre
                                class="line before"><span class="ws">        </span>packet = packet_type(bytes(buff), self.encoding)</pre>
                            <pre class="line before"><span class="ws">        </span>if packet.is_error_packet():</pre>
                            <pre
                                class="line before"><span class="ws">            </span>if self._result is not None and self._result.unbuffered_active is True:</pre>
                            <pre
                                class="line before"><span class="ws">                </span>self._result.unbuffered_active = False</pre>
                            <pre class="line current"><span class="ws">            </span>packet.raise_for_error()</pre>
                            <pre class="line after"><span class="ws">        </span>return packet</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                            <pre class="line after"><span class="ws">    </span>def _read_bytes(self, num_bytes):</pre>
                            <pre
                                class="line after"><span class="ws">        </span>self._sock.settimeout(self._read_timeout)</pre>
                            <pre class="line after"><span class="ws">        </span>while True:</pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916739920">
                        <h4>File
                            <cite class="filename">"/usr/local/lib/python3.9/site-packages/pymysql/protocol.py"</cite>,
                            line <em class="line">219</em>,
                            in <code class="function">raise_for_error</code></h4>
                        <div class="source library">
                            <pre class="line before"><span class="ws">        </span>self.rewind()</pre>
                            <pre
                                class="line before"><span class="ws">        </span>self.advance(1)  # field_count == error (we already know that)</pre>
                            <pre class="line before"><span class="ws">        </span>errno = self.read_uint16()</pre>
                            <pre class="line before"><span class="ws">        </span>if DEBUG:</pre>
                            <pre
                                class="line before"><span class="ws">            </span>print(&#34;errno =&#34;, errno)</pre>
                            <pre
                                class="line current"><span class="ws">        </span>err.raise_mysql_exception(self._data)</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                            <pre class="line after"><span class="ws">    </span>def dump(self):</pre>
                            <pre class="line after"><span class="ws">        </span>dump_packet(self._data)</pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                            <pre class="line after"><span class="ws"></span> </pre>
                        </div>
                    </div>

                <li>
                    <div class="frame" id="frame-140118916776560">
                        <h4>File <cite class="filename">"/usr/local/lib/python3.9/site-packages/pymysql/err.py"</cite>,
                            line <em class="line">150</em>,
                            in <code class="function">raise_mysql_exception</code></h4>
                        <div class="source library">
                            <pre class="line before"><span class="ws">    </span>else:</pre>
                            <pre
                                class="line before"><span class="ws">        </span>errval = data[3:].decode(&#34;utf-8&#34;, &#34;replace&#34;)</pre>
                            <pre class="line before"><span class="ws">    </span>errorclass = error_map.get(errno)</pre>
                            <pre class="line before"><span class="ws">    </span>if errorclass is None:</pre>
                            <pre
                                class="line before"><span class="ws">        </span>errorclass = InternalError if errno &lt; 1000 else OperationalError</pre>
                            <pre class="line current"><span class="ws">    </span>raise errorclass(errno, errval)</pre>
                        </div>
                    </div>
            </ul>
            <blockquote>pymysql.err.OperationalError: (1054, &#34;Unknown column &#39;reset_token&#39; in &#39;field
                list&#39;&#34;)
            </blockquote>
        </div>

        <div class="plain">
            <p>
                This is the Copy/Paste friendly version of the traceback.
            </p>
            <textarea cols="50" rows="10" name="code" readonly>Traceback (most recent call last):
  File &#34;/usr/local/lib/python3.9/site-packages/flask/app.py&#34;, line 1536, in __call__
    return self.wsgi_app(environ, start_response)
  File &#34;/usr/local/lib/python3.9/site-packages/flask/app.py&#34;, line 1514, in wsgi_app
    response = self.handle_exception(e)
  File &#34;/usr/local/lib/python3.9/site-packages/flask/app.py&#34;, line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File &#34;/usr/local/lib/python3.9/site-packages/flask/app.py&#34;, line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &#34;/usr/local/lib/python3.9/site-packages/flask/app.py&#34;, line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File &#34;/usr/local/lib/python3.9/site-packages/flask/app.py&#34;, line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File &#34;/app/routes/app/forgot_password.py&#34;, line 18, in forgot_password
    cursor.execute(
  File &#34;/usr/local/lib/python3.9/site-packages/pymysql/cursors.py&#34;, line 153, in execute
    result = self._query(query)
  File &#34;/usr/local/lib/python3.9/site-packages/pymysql/cursors.py&#34;, line 322, in _query
    conn.query(q)
  File &#34;/usr/local/lib/python3.9/site-packages/pymysql/connections.py&#34;, line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File &#34;/usr/local/lib/python3.9/site-packages/pymysql/connections.py&#34;, line 825, in _read_query_result
    result.read()
  File &#34;/usr/local/lib/python3.9/site-packages/pymysql/connections.py&#34;, line 1199, in read
    first_packet = self.connection._read_packet()
  File &#34;/usr/local/lib/python3.9/site-packages/pymysql/connections.py&#34;, line 775, in _read_packet
    packet.raise_for_error()
  File &#34;/usr/local/lib/python3.9/site-packages/pymysql/protocol.py&#34;, line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File &#34;/usr/local/lib/python3.9/site-packages/pymysql/err.py&#34;, line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.OperationalError: (1054, &#34;Unknown column &#39;reset_token&#39; in &#39;field list&#39;&#34;)
</textarea>
        </div>
        <div class="explanation">
            The debugger caught an exception in your WSGI application. You can now
            look at the traceback which led to the error. <span class="nojavascript">
  If you enable JavaScript you can also use additional features such as code
  execution (if the evalex feature is enabled), automatic pasting of the
  exceptions and much more.</span>
        </div>
        <div class="footer">
            Brought to you by <strong class="arthur">DON'T PANIC</strong>, your
            friendly Werkzeug powered traceback interpreter.
        </div>
    </div>

    <div class="pin-prompt">
        <div class="inner">
            <h3>Console Locked</h3>
            <p>
                The console is locked and needs to be unlocked by entering the PIN.
                You can find the PIN printed out on the standard output of your
                shell that runs the server.
            <form>
                <p>PIN:
                    <input type=text name=pin size=14>
                    <input type=submit name=btn value="Confirm Pin">
            </form>
        </div>
    </div>
</body>

</html>

<!--

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1536, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1514, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/app/routes/app/forgot_password.py", line 18, in forgot_password
    cursor.execute(
  File "/usr/local/lib/python3.9/site-packages/pymysql/cursors.py", line 153, in execute
    result = self._query(query)
  File "/usr/local/lib/python3.9/site-packages/pymysql/cursors.py", line 322, in _query
    conn.query(q)
  File "/usr/local/lib/python3.9/site-packages/pymysql/connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/usr/local/lib/python3.9/site-packages/pymysql/connections.py", line 825, in _read_query_result
    result.read()
  File "/usr/local/lib/python3.9/site-packages/pymysql/connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
  File "/usr/local/lib/python3.9/site-packages/pymysql/connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "/usr/local/lib/python3.9/site-packages/pymysql/protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/usr/local/lib/python3.9/site-packages/pymysql/err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.OperationalError: (1054, "Unknown column 'reset_token' in 'field list'")


-->
```
and on the server this:
```bash
INFO:werkzeug:172.18.0.1 - - [28/Sep/2025 16:58:22] "POST /forgot_password HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1536, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1514, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/app/routes/app/forgot_password.py", line 18, in forgot_password
    cursor.execute(
  File "/usr/local/lib/python3.9/site-packages/pymysql/cursors.py", line 153, in execute
    result = self._query(query)
  File "/usr/local/lib/python3.9/site-packages/pymysql/cursors.py", line 322, in _query
    conn.query(q)
  File "/usr/local/lib/python3.9/site-packages/pymysql/connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/usr/local/lib/python3.9/site-packages/pymysql/connections.py", line 825, in _read_query_result
    result.read()
  File "/usr/local/lib/python3.9/site-packages/pymysql/connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
  File "/usr/local/lib/python3.9/site-packages/pymysql/connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "/usr/local/lib/python3.9/site-packages/pymysql/protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/usr/local/lib/python3.9/site-packages/pymysql/err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.OperationalError: (1054, "Unknown column 'reset_token' in 'field list'")
```

- when i do forgot password:
on postman
```html
<!doctype html>
<html lang=en>
<title>415 Unsupported Media Type</title>
<h1>Unsupported Media Type</h1>
<p>Did not attempt to load JSON data because the request Content-Type was not &#39;application/json&#39;.</p>
```

ErwannL reviewed (changes requested):
can you please check the errors i get ans solve them

ErwannL reviewed (changes requested):
After a discossion on discord, i had explain to you that previous errors were due to :
- an error on my side for the db
- an error on the Postman collection

both are resolved

Now can you please add that the token for reseting the password is destroyed after a certain amount of time ?

ErwannL committed:
Refactoring verification of databases to check the tables instead of just a ping

ErwannL commented:
I have a question, why and where did you use FLASK_SECRET_KEY ?

ErwannL committed:
Merge pull request #326 from Jessika63/BACK_verify-tables-in-dbs

Back verify tables in dbs

ErwannL committed:
Add QR code CRUD endpoints and DB tables

Introduces endpoints for generating, reading, and deleting prescription, direction, and profile QR codes. Adds new database tables for qrcodes_maps and qrcodes_profiles, and updates logic to store and retrieve QR code data from the database. Endpoints now support one-time use and role-based profile data filtering.

ErwannL committed:
Merge branch 'BACK_update-qrcode-by-using-db' into APP_add_auth_routes

TurpaultM committed:
UPDATE: Remove token after reset

TurpaultM committed:
Merge branch 'APP_add_auth_routes' of github.com:Jessika63/PharmaXcess into APP_add_auth_routes

TurpaultM commented:
> After a discossion on discord, i had explain to you that previous errors were due to :
> 
> * an error on my side for the db
> * an error on the Postman collection
> 
> both are resolved
> 
> Now can you please add that the token for reseting the password is destroyed after a certain amount of time ?

done

TurpaultM commented:
> I have a question, why and where did you use FLASK_SECRET_KEY ?

I use it in the .env backend to connect a user, the .env is on discord

ErwannL reviewed (changes requested):
if i register
then forget password a token is generated 
but if i login after the token stay here 

can you please delete it if it is here when you log in

ErwannL committed:
Update db connection import to get_app_connection

Replaced all imports of get_connection with get_app_connection in pharmacy and QR code route files for consistency with updated database access method.

TurpaultM committed:
UPDATE: Remove token after login
